### PR TITLE
feat: add completions to spacectl

### PR DESCRIPTION
**Pull request by znd4**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sp/spacectl/package.nix [hunk 1]
```diff
@@ -17,6 +17,13 @@ buildGoModule rec {
 
   vendorHash = "sha256-SYfXG6YM0Q2rCnoTM2tYvE17uBCD8yQiW/5DTCxMPWo=";
 
+  postInstall = ''
```
Comment (getchoo) [inline pkgs/by-name/sp/spacectl/package.nix:22]: We're running the built binary to generate these completions, so we need to make sure the system we're building on can execute code meant for the system it's building for (i.e., when cross compiling for aarch64-linux, can x86_64-linux run that code?)

We can do this with the following conditional
```suggestion
  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```

File: pkgs/by-name/sp/spacectl/package.nix [hunk 2]
```diff
@@ -17,6 +17,15 @@ buildGoModule rec {
 
   vendorHash = "sha256-SYfXG6YM0Q2rCnoTM2tYvE17uBCD8yQiW/5DTCxMPWo=";
 
+  nativeBuildInputs = [installShellFiles];
```
Comment (getchoo) [inline pkgs/by-name/sp/spacectl/package.nix:20]:
```suggestion
  nativeBuildInputs = [ installShellFiles ];
```

To conform with `nixfmt`

**Review by getchoo — CHANGES_REQUESTED** [review]
Change itself looks good, only have the notes below. I would also review our commit conventions in the [contributing guide](https://github.com/NixOS/nixpkgs/blob/fc1a0e3323efa875a3be77730bddfcd03f53175e/pkgs/README.md#commit-conventions) -- a better title for this may be something like `spacectl: add completions`

And welcome to Nixpkgs! :)

File: pkgs/by-name/sp/spacectl/package.nix (hunk 1 above)
Comment (znd4) [reply pkgs/by-name/sp/spacectl/package.nix:23]: 😮

File: pkgs/by-name/sp/spacectl/package.nix [hunk 3]
```diff
@@ -2,6 +2,7 @@
   lib,
```
Comment (getchoo) [inline pkgs/by-name/sp/spacectl/package.nix:2]:
```suggestion
  lib,
  stdenv,
```

**Review by getchoo — CHANGES_REQUESTED** [review]
Should be good after this :p

File: pkgs/by-name/sp/spacectl/package.nix (hunk 3 above)
Comment (znd4) [reply pkgs/by-name/sp/spacectl/package.nix:2]: oops, sorry I should've left this as draft while I worked through these things

File: pkgs/by-name/sp/spacectl/package.nix [hunk 4]
```diff
@@ -17,6 +19,15 @@ buildGoModule rec {
...
   vendorHash = "sha256-SYfXG6YM0Q2rCnoTM2tYvE17uBCD8yQiW/5DTCxMPWo=";
 
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
+    installShellCompletion --cmd spacectl \
+      --bash <($out/bin/spacectl completion bash) \
+      --fish <($out/bin/spacectl completion fish) \
+      --zsh <($out/bin/spacectl completion zsh) \
+  '';
```
[the remark this answers is not in the document]
Comment (znd4) [reply pkgs/by-name/sp/spacectl/package.nix:29]: 🤔 I don't understand this, but I see that it's used in a lot of other places, so I'll just go along with it

**Review by getchoo — APPROVED** [review]

File: pkgs/by-name/sp/spacectl/package.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (ambroisie) [reply pkgs/by-name/sp/spacectl/package.nix]: When cross-compiling, you might not always be able to execute the compiled code (e.g: linux-aarch64 on linux-x86_64). We can use `qemu`, `wine`, `wasmi` etc... to run code for the target platform on the host platform. That's what `stdenv.hostPlatform.emulator` does: it selects the appropriate package for emulating the target platform, given the set of packages that can run on the build host.

[the remark this answers is not in the document]
Comment (znd4) [reply pkgs/by-name/sp/spacectl/package.nix]: ooh, nifty! thanks for the explainer :)

**Pull request merged**